feat(core): add automated migration to replace async with waitForAsync

crisbeto opened the pull request:
Adds a migration that finds all imports and calls to the deprecated `async` function from `@angular/core/testing` and replaces them with `waitForAsync`.

These changes also move a bit of code out of the `Renderer2` migration so that it can be reused.

crisbeto commented on packages/core/schematics/tsconfig.json:
```diff
@@ -4,7 +4,7 @@
     "strict": true,
     "lib": ["es2015"],
-    "types": [],
+    "types": ["jasmine"],
```
This change isn't required for the migration, but it makes it so that vscode can autocomplete the Jasmine functions in tests.

josephperrott commented on packages/core/schematics/migrations/wait-for-async/README.md:
```diff
@@ -0,0 +1,21 @@
+## async -> waitForAsync migration
+
+Automatically migrates from `async` to `wait` by changing function calls and renaming imports.
```
```suggestion
Automatically migrates from `async` to `waitForAsync` by changing function calls and renaming imports.
```

josephperrott reviewed (approved):
LGTM

Just the one comment to take a look at.

crisbeto committed:
feat(core): add automated migration to replace async with waitForAsync

Adds a migration that finds all imports and calls to the deprecated `async` function from
`@angular/core/testing` and replaces them with `waitForAsync`.

These changes also move a bit of code out of the `Renderer2` migration so that it can be reused.
